book/src/dev/rfcs/xxxx-block-subsidy.md

```diff
@@ -0,0 +1,313 @@
+[reference-level-explanation]: #reference-level-explanation
+
+Given the module structure proposed above all the block subsidy calculations from the protocol must be implemented. The final goal is to do checks for each incoming block and make sure they pass the consensus rules.
+
+To do the calculations and checks the following constants, types and functions need to be introduced:
+
+## Constants
+
+- `SLOW_START_INTERVAL: Height`
+- `SLOW_START_SHIFT: Height`
+- `MAX_BLOCK_SUBSIDY: u64`
+- `PRE_BLOSSOM_POW_TARGET_SPACING: Duration`
+- `POST_BLOSSOM_POW_TARGET_SPACING: Duration`
+- `BLOSSOM_POW_TARGET_SPACING_RATIO: u64`
+- `PRE_BLOSSOM_HALVING_INTERVAL: Height`
+- `POST_BLOSSOM_HALVING_INTERVAL: Height`
+- `FOUNDERS_FRACTION_DIVISOR: u64`
+- `FOUNDERS_ADDRESS_COUNT: u32`
+- `FOUNDER_ADDRESSES_MAINNET: [&str; 48]`
+- `FOUNDER_ADDRESSES_TESTNET: [&str; 48]`
+
+### Funding streams parameter constants
+
+The design suggests to implement the parameters needed for funding streams as:
+
+```
+/// The funding stream receiver categories
+#[allow(missing_docs)]
+pub enum FundingStreamReceiver {
+    BootstrapProject,
+    ZcashFoundation,
+    MajorGrants,
+}
+
+/// The numerator for each funding stream receiving category 
+/// as described in [protocol specification §7.9.1][7.9.1].
+/// 
+/// [7.9.1]: https://zips.z.cash/protocol/protocol.pdf#zip214fundingstreams
+const FUNDING_STREAM_RECEIVER_NUMERATORS: &[(FundingStreamReceiver, u64)] = &[
+    (FundingStreamReceiver::BootstrapProject, 7),
+    (FundingStreamReceiver::ZcashFoundation, 5),
+    (FundingStreamReceiver::MajorGrants, 8),
+];
+
+/// Denominator as described in [protocol specification §7.9.1][7.9.1].
+/// 
+/// [7.9.1]: https://zips.z.cash/protocol/protocol.pdf#zip214fundingstreams
+pub const FUNDING_STREAM_RECEIVER_DENOMINATOR: u64 = 100;
+
+#[allow(missing_docs)]
+pub enum FundingStreamRange {
+    StartHeight,
+    EndHeight,
+}
+
+/// Start and end Heights for funding streams 
+/// as described in [protocol specification §7.9.1][7.9.1].
+/// 
+/// [7.9.1]: https://zips.z.cash/protocol/protocol.pdf#zip214fundingstreams
+const FUNDING_STREAM_HEIGHT_RANGES: &[(Network, Range<_>)] = &[
+    (Network::Mainnet, Height(1_046_400)..Height(2_726_400)),
+    (Network::Testnet, Height(1_028_500)..Height(2_796_000)),
+];
+```
+
+Each `FundingStreamReceiver` for the mainnet must have 48 payment addresses associated with and 51 for the testnet. 
+
+https://zips.z.cash/zip-0214#mainnet-recipient-addresses
+
+The following constants are needed:
+
+- `FUNDING_STREAM_BP_ADDRESSES_MAINNET: [&str; 48]`
+- `FUNDING_STREAM_ZF_ADDRESSES_MAINNET: [&str; 48]`
+- `FUNDING_STREAM_MG_ADDRESSES_MAINNET: [&str; 48]`
+- `FUNDING_STREAM_BP_ADDRESSES_TESTNET: [&str; 51]`
+- `FUNDING_STREAM_ZF_ADDRESSES_TESTNET: [&str; 51]`
+- `FUNDING_STREAM_MG_ADDRESSES_TESTNET: [&str; 51]`
+
+## General subsidy
+
+The block subsidy and miner reward among other utility functions are inside the general subsidy category.
```

Comment:
The miner reward is a contextual validation check, so it doesn't belong in this semantic verification RFC.

Contextual validation is implemented in `zebra-state`, so we can't specify it as part of the general module in `zebra-consensus`.

```suggestion
The block subsidy and other utility functions are inside the general subsidy category.
```